x/evm/keeper/state_transition.go

```diff
@@ -67,7 +67,15 @@ func (k Keeper) GetHashFn() vm.GetHashFunc {
 		case k.ctx.BlockHeight() == h:
 			// Case 1: The requested height matches the one from the context so we can retrieve the header
 			// hash directly from the context.
-			return common.BytesToHash(k.ctx.HeaderHash())
+			// Note: we cannot use k.ctx.HeaderHash() because the headerHash it set only at begin block
+			// so it won't be set in case of simulated context
```

Comment:
isn't it set for checkTxState too? can you open an issue on the SDK to fix this and include it on v0.43?

Reply:
It is also set for checkContext.  
I realised my comment might be confusing.  What I want to mean is that in case of a "query" context, such as simulating a transaction through "EthCall", then the header hash won't be set.
I have rewrote the comment, let me know if its clear now